Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -828,11 +829,21 @@ void ContainerNode::removeChildren()
     Ref protectedThis { *this };
     NodeVector removedChildren;
     removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren);
-
+    delayDeletingRemovedChildren(removedChildren);
```

Comment:
This is problematic. This code would mean that we'd continue to accumulate nodes in cases like this:
```js
for (let i = 0; i < 100000; ++i)
    document.body.innerHTML = '<span>some node</span>';
```

Reply:
I see the logic of this optimization, but I worry that there's no practical way to make it work.

When the count of nodes is small, it's OK to delay destruction, but not profitable.

When the count of nodes is large, it's profitable to delay destruction, but not OK.